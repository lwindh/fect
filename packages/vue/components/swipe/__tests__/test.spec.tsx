import { flushPromises, mount } from '@vue/test-utils'
import Swipe from '..'
import SwipeItem from '../../swipe-item'

describe('Swipe', () => {
  it('should be support custom render indicator', async () => {
    const slots = {
      indicator: () => <div>custom</div>,
    }

    const wrapper = mount({
      render() {
        return (
          <Swipe v-slots={slots}>
            <SwipeItem>1</SwipeItem>
            <SwipeItem>2</SwipeItem>
          </Swipe>
        )
      },
    })
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
