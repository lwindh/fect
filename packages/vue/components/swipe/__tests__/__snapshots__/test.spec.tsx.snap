// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swipe should be support custom render indicator 1`] = `
<div class="fect-swipe">
  <div class="fect-swipe__track" style="width: 0px; transform: translateX(0px); transition-duration: 0ms;">
    <div class="fect-swipe-item" style="width: 0px; transform: translateX(0px);">1</div>
    <div class="fect-swipe-item" style="width: 0px; transform: translateX(0px);">2</div>
  </div>
  <div>custom</div>
</div>
`;
